lib/nvmf/subsystem.c

```diff
@@ -338,6 +338,16 @@ spdk_nvmf_subsystem_create(struct spdk_nvmf_tgt *tgt,
 	return subsystem;
 }
 
+/* Register a callback function for the subsystem events*/
+int
+spdk_nvmf_subsystem_register_for_event(struct spdk_nvmf_subsystem *subsystem,
+						spdk_nvmf_subsystem_event_cb cb)
+{
+	subsystem->nvmf_ss_event_cb = cb;
```

Comment:
What is the intended usage for this callback?

Reply:
nexus is exported as nvmf target by creating a subsystem using spdk_nvmf_subsystem_create.

We need an additional API from spdk something like spdk_nvmf_subsystem_register_for_event(struct spdk_nvmf_subsystem *subsystem, func *cb, void *args) And once it is registered for events, it should call the callback function whenever any initiator connects or disconnects/timeout to the given subsystem

Reply:
I mean what we'd do with the callback, is this for logging/tracing, to notify the control-plane etc?

Reply:
ah, ok! This is to help fix https://dcsw-bolt.atlassian.net/browse/B2-743 to avoid data corruption due to old IOs staying in the replica connections. I have made the API generic such that it may be useful in some other scenarios as well